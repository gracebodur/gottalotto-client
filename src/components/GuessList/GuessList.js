import React, { Component } from 'react'
import Guess from '../Guess/Guess'
import GuessListContext from '../../contexts/guessListContext'
import GuessesApiService from '../../services/guesses-api-service'

class GuessList extends Component {
	static contextType = GuessListContext;
	componentDidMount() {
		this.context.clearError();
		GuessesApiService.getAllGuesses()
			.then(this.context.setGuessList)
			.catch(this.context.setError);
	}
	renderGuesses() {
		const { guessList = [] } = this.context;
		return guessList.map(guess => (
			<Guess key={guess.guess_id} guess={guess} />
		));
	}
	render() {
		const { error } = this.context;
		return (
			<section>
				<h2>Everyone else's guesses:</h2>
				{/* <div className='winnerList'>
					<div className='guessItem container'>
						<img src='https://via.placeholder.com/75' />
						<h3 className='guessName'>Grace Bodur</h3>
						<h3 className='guessNumbers'>05, 08, 16, 18, 26, 09</h3>
						<p className='guessMessage'>If you have a few dollars to spare, donate to this wonderful cause: https://comngood.org/home</p>
					</div>
					<div className='guessItem container'>
						<img src='https://via.placeholder.com/75' />
						<h3 className='guessName'>Malek Haj-Hussein</h3>
						<h3 className='guessNumbers'>06, 09, 16, 19, 29, 55</h3>
						<p className='guessMessage'>Check out my social media pages! https://www.linkedin.com/in/malekadair/</p>
					</div>
					<div className='guessItem container'>
						<img src='https://via.placeholder.com/75' />
						<h3 className='guessName'>Gary</h3>
						<h3 className='guessNumbers'>07, 07, 07, 07, 07, 07</h3>
						<p className='guessMessage'>A'jahn is cool.</p>
					</div>
				</div> */}
				{error ? (
					<p className="red">There was an error, try again</p>
				) : (
						this.renderGuesses()
					)}
				<a href='https://www.powerball.com/games/home'><p>Power Ball Official Website</p></a>
			</section>
		)
	}
}

export default GuessList