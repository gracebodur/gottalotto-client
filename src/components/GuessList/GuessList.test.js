import React from "react";
import toJson from "enzyme-to-json";
import GuessList from "./GuessList";

describe(`GuessList component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a GuessList by default", () => {
		const wrapper = <GuessList />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the GuessList given props", () => {
		const wrapper = <GuessList {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});