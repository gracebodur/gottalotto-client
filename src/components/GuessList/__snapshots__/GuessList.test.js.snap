// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessList component renders a GuessList by default 1`] = `undefined`;

exports[`GuessList component renders the GuessList given props 1`] = `undefined`;
